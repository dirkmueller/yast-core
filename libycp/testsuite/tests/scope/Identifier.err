Parsed:
----------------------------------------------------------------------
{
    module "test";
}
{
    // integer a
    integer a = 1;
    global integer this_is_global = 2;
}
----------------------------------------------------------------------
[Scanner] tests/Identifier.ycp:32 Warning: Definition shadows global symbol
[Scanner] tests/Identifier.ycp:26 Warning: 'a' defined here.
Parsed:
----------------------------------------------------------------------
{
    global integer a = 55;
    return {
    // integer a
    integer a = 42;
    return [test::this_is_global, ::a, a];
}
;
}
----------------------------------------------------------------------
