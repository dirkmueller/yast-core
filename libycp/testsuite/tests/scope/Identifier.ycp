#
# ---------------------------------------------------------
#
#  Filename:    Identifier.ycp
#
#  Purpose:     test identifier ycp type
#
#  Creator:     kkaempf@suse.de
#
#  Maintainer:  kkaempf@suse.de
#
# ---------------------------------------------------------


{
  module "test";

  export this_is_global;

  integer a = 1;
  global integer this_is_global = 2;
  // implicit: return nil;
}

{
  global integer a = 55;	// this is accessible via ::a

  // should be [2, 55, 42]
  //   and a warning about local a shadowing global a

  return {   import "test";
	integer a = 42; return [ test::this_is_global, ::a, a ]; };
}
