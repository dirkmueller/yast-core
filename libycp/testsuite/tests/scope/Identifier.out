(nil)
([2, 55, 42])
